pkg/agent/tunnel/tunnel.go

```diff
@@ -41,6 +42,11 @@ func getAddresses(endpoint *v1.Endpoints) []string {
 		}
 		for _, address := range subset.Addresses {
 			serverAddress := address.IP
+			// Assuming address is a literal IPv6 if it contains colons;
```

Comment:
Can you just use net.JoinHostPort
```golang
if port != "" {
  serverAddress = net.JoinHostPort(serverAddress, port)
}
```

Reply:
But an IPv6 address must be enclosed with the `[]` even if port is empty, doesn't it?

Reply:
Please look at the description and source of `JoinHostPort` @narqo (https://golang.org/src/net/ipsock.go?s=6810:6853#L217), appears to be very close to the logic you have here.

Reply:
Ack, very sorry, I misread the question @narqo, too early need coffee. ;) 

If we just `serverAddress = net.JoinHostPort(serverAddress, port)` does that work? 
I think we should always have a port there, and maybe default it if not, as the load balancer would need it.

Reply:
Works for me. 🆙